R/zchunk_LA116.geo.R

```diff
@@ -1,23 +1,27 @@
 #' module_energy_LA116.geo
 #'
-#' Briefly describe what this chunk does.
+#' Generate geothermal (hydrothermal and engineered geothermal systems (EGS)) supply curves by GCAM region
+#' Supply curves developed for the 14 GCAM 3.0 regions are downscaled to the country level on the basis
+#' of land area, and aggregated to the current GCAM regions. The data source for the supply curves is
+#' Hannam et al. 2009: http://www.pnl.gov/main/publications/external/technical_reports/PNNL-19231.pdf
 #'
 #' @param command API command to execute
 #' @param ... other optional parameters, depending on command
 #' @return Depends on \code{command}: either a vector of required inputs,
 #' a vector of output names, or (if \code{command} is "MAKE") all
 #' the generated outputs: \code{L116.RsrcCurves_EJ_R_geo}, \code{L116.RsrcCurves_EJ_R_EGS}. The corresponding file in the
 #' original data system was \code{LA116.geo.R} (energy level1).
-#' @details Describe in detail what this chunk does.
+#' @details This code chunk assigns geothermal electric resources to GCAM regions. Because the underlying
+#' source data were developed for the 14 regions of GCAM 3.0 and prior, they are first
+#' downscaled to the nation level on the basis of land area, and aggregated to the current GCAM regions.
 #' @importFrom assertthat assert_that
 #' @importFrom dplyr filter mutate select
 #' @importFrom tidyr gather spread
-#' @author YourInitials CurrentMonthName 2017
-#' @export
-module_energy_LA116.geo_DISABLED <- function(command, ...) {
+#' @author GPK April 2017
+module_energy_LA116.geo <- function(command, ...) {
   if(command == driver.DECLARE_INPUTS) {
     return(c(FILE = "common/iso_GCAM_regID",
-             "L100.Land_type_area_ha",
+             FILE = "aglu/LDS/Land_type_area_ha",
```

Comment:
BTW normally we would use the output of the L100 script, because the data are cleaned a bit, but there are several reasons why that is not done here. For one, in the old data system, I didn't want for the LA116 energy file to be dependent on a level1 aglu file, because that would make for more back-and-forth in the hard-wired dependencies set in the makefile (i.e., would need to run LA100 ag before running LA116 energy, which currently isn't the file run order). Two, one of the steps in the LA100 aglu code chunk that pre-processes the LDS data is to map Taiwan back to China, as Taiwan is not an aglu region. However, it is a region in the energy system, so it's actually preferable to keep it separate, as it is when written out by the LDS. And third, this is the way to replicate the data in the old data system.